torch/csrc/jit/passes/remove_mutation.h

```diff
@@ -1,12 +1,270 @@
 #pragma once
 
+#include <c10/util/Exception.h>
 #include <torch/csrc/WindowsTorchApiMacro.h>
+#include <torch/csrc/jit/ir/alias_analysis.h>
 #include <torch/csrc/jit/ir/ir.h>
+#include <torch/csrc/utils/memory.h>
 
 namespace torch {
 namespace jit {
 
-struct MutationRemover;
+struct MutationRemover {
```

Comment:
Can you move some of the implementation out of the header, or use an impl pattern? Having the whole struct in the .h and then four lines of codes in the cpp file will lead to bad bulid times when I'm working on this file. 

Another, maybe easiest alternative is to add a `inplaceOpVariant` api that takes in an AliasDb and a node.
```
void InplaceOpVariant(Node *n, const AliasDb& aliasDb)
```

and then have mutation remover / your pass both invoke that